core/src/main/java/org/apache/calcite/sql/type/ArrayElementOperandTypeChecker.java

```diff
@@ -31,11 +32,41 @@
  * Parameter type-checking strategy where types must be Array and Array element type.
  */
 public class ArrayElementOperandTypeChecker implements SqlOperandTypeChecker {
+  //~ Instance fields --------------------------------------------------------
+
+  private final boolean allowNullCheck;
+  private final boolean allowCast;
```

Comment:
if this is always "false", maybe it shouldn't be actually a field until there is some use for it.

Reply:
Here, by adding member variables, these two parameters can be passed in through different constructors. For functions that do not require null checking, use the default constructor. For scenarios that require null checking or allow cast, it will be more versatile to pass them in through different constructors. what do you think?

Reply:
I understand why, but I am pointing out that all uses of this class now have allowCast = false.
Do you foresee a use case where allowCast = true?

Reply:
Not currently, but since it is a public operand checker class, it may be used for other purposes in the future. I would be ok if I set it directly to false.

thanks for patient reviewing.